CMakeLists.txt

```diff
@@ -135,6 +146,11 @@ FOREACH(SUFFIX "_DEBUG" "_RELEASE" "_RELWITHDEBINFO" "_MINSIZEREL" "")
   set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY${SUFFIX} "${PROJECT_BINARY_DIR}/lib")
 ENDFOREACH()
 
+
+if(BYN_ENABLE_LTO)
+  add_compile_flag("-flto=thin")
```

Comment:
does this work with MSVC?

Reply:
Oh, it works with clang-cl but probably not with MSVC. There must be some good way to tell the difference but I'll have to look into what it is (I think the `MSVC` variable in CMake is true when using clang-cl because it accepts MSVC-style flags).

Reply:
What about gcc?

Reply:
It does look like `-flto=thin` is not supported in gcc AFAICT, but `-flto` is.

Is `-flto` too slow on binaryen?

Reply:
In answer to this, I think I'd prefer to use the same LTO mechanism in LLVM and Binaryen for now to keep our support matrix small. supporting regular LTO and gcc (and MSVC's LTCG)  would probably work but will complicate this code. We can add those later if people want them.